netket/vqs/full_summ/expect.py

```diff
@@ -79,7 +83,7 @@ def expect_and_forces(
 
     O = sparsify(Ô)
     Ψ = vstate.to_array()
-    OΨ = O @ Ψ
+    OΨ = jax.jit(lambda x, y: x @ y)(O, Ψ)  # jit for gda
```

Comment:
with gda i refer to global device arrays (which have been merged into jax.Array, [see docs](https://jax.readthedocs.io/en/latest/jax_array_migration.html#new-public-type-named-jax-array))

it's to avoid this error:
```RuntimeError: Running operations on `Array`s that are not fully addressable by this process (i.e. `Array`s with data sharded across multiple devices and processes.) is dangerous. It’s very important that all processes run the same cross-process computations in the same order otherwise it can lead to hangs. If you’re not already familiar with JAX’s multi-process programming model, please read https://jax.readthedocs.io/en/latest/multi_process.html. To fix this error, run your `jitted` computation inside `with jax.spmd_mode('allow_all'):` context manager.```

I guess I could have used the [context manager](https://github.com/google/jax/blob/57e33dc3b553fc49095090d0389b7cbf0e694728/jax/_src/config.py#L847-L857) here, so far I have just added jit around everything

Reply:
@inailuig didn't we agree that O is just a fully replicated matrix for now, as should be psi, in fullsumstate, for now?

Reply:
I changed it so that I just extract one local shard from the (replicated) global array, now I just do everything like in the scalar case.